feature/qos/ate_tests/qos_policy_config_test/qos_policy_config_test.go

```diff
@@ -1146,3 +1164,580 @@ func testCiscoSchedulerPoliciesConfig(t *testing.T) {
+		dropEnabled               bool
+		minThreshold              uint64
+		maxThreshold              uint64
+		maxDropProbabilityPercent uint8
+		weight                    uint32
+	}{
+		ecnEnabled:                true,
+		dropEnabled:               false,
+		minThreshold:              uint64(80000),
+		maxThreshold:              math.MaxUint64,
+		maxDropProbabilityPercent: uint8(1),
+		weight:                    uint32(0),
+	}
+
+	queueMgmtProfile := q.GetOrCreateQueueManagementProfile("DropProfile")
+	queueMgmtProfile.SetName("DropProfile")
+	wred := queueMgmtProfile.GetOrCreateWred()
+	uniform := wred.GetOrCreateUniform()
+	uniform.SetEnableEcn(ecnConfig.ecnEnabled)
+	uniform.SetMinThreshold(ecnConfig.minThreshold)
+	uniform.SetMaxThreshold(ecnConfig.maxThreshold)
+	uniform.SetMaxDropProbabilityPercent(ecnConfig.maxDropProbabilityPercent)
+
+	t.Logf("qos ECN QueueManagementProfile config cases: %v", ecnConfig)
+	gnmi.Replace(t, dut, gnmi.OC().Qos().Config(), q)
+
+	// Verify the QueueManagementProfile is applied by checking the telemetry path state values.
+	wredUniform := gnmi.OC().Qos().QueueManagementProfile("DropProfile").Wred().Uniform()
+
+	if got, want := gnmi.Get(t, dut, wredUniform.MaxDropProbabilityPercent().State()), ecnConfig.maxDropProbabilityPercent; got != want {
+		t.Errorf("wredUniform.MaxDropProbabilityPercent().State(): got %v, want %v", got, want)
+	}
+
+	if !deviations.StatePathsUnsupported(dut) {
+		if got, want := gnmi.Get(t, dut, wredUniform.MinThreshold().State()), ecnConfig.minThreshold; got != want {
+			t.Errorf("wredUniform.MinThreshold().State(): got %v, want %v", got, want)
+		}
+		if got, want := gnmi.Get(t, dut, wredUniform.MaxThreshold().State()), ecnConfig.maxThreshold; got != want {
+			t.Errorf("wredUniform.MaxThreshold().State(): got %v, want %v", got, want)
+		}
+	}
+
+	if !deviations.DropWeightLeavesUnsupported(dut) {
+
+		uniform.SetDrop(ecnConfig.dropEnabled)
+		uniform.SetWeight(ecnConfig.weight)
+		gnmi.Replace(t, dut, gnmi.OC().Qos().Config(), q)
+
+		if got, want := gnmi.Get(t, dut, wredUniform.Drop().State()), ecnConfig.dropEnabled; got != want {
+			t.Errorf("wredUniform.Drop().State(): got %v, want %v", got, want)
+		}
+		if got, want := gnmi.Get(t, dut, wredUniform.Weight().State()), ecnConfig.weight; got != want {
+			t.Errorf("wredUniform.Weight().State(): got %v, want %v", got, want)
+		}
+	}
+
+	cases := []struct {
+		desc        string
+		targetGroup string
+		ecnProfile  string
+		scheduler   string
+	}{{
+		desc:        "output-interface-BE1",
+		targetGroup: "BE1",
+		ecnProfile:  "DropProfile",
+		scheduler:   "scheduler",
+	}, {
+		desc:        "output-interface-BE0",
+		targetGroup: "BE0",
+		ecnProfile:  "DropProfile",
+		scheduler:   "scheduler",
+	}, {
+		desc:        "output-interface-AF1",
+		targetGroup: "AF1",
+		ecnProfile:  "DropProfile",
+		scheduler:   "scheduler",
+	}, {
+		desc:        "output-interface-AF2",
+		targetGroup: "AF2",
+		ecnProfile:  "DropProfile",
```

Comment:
ecnProfile and scheduler are not parameterized, so they can be inlined

Reply:
addressed